pkg/authorization/controller/authorizationsync/normalize.go

```diff
@@ -3,16 +3,141 @@ package authorizationsync
 import (
 	"strings"
 
-	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
-
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/apis/rbac"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
 )
 
-// NormalizePolicyRules mutates the given rules and lowercases verbs, resources and API groups.
+func PrepareForCreateClusterRole(originClusterRole *authorizationapi.ClusterRole) (*rbac.ClusterRole, error) {
+	// convert the origin role to an rbac role and compare the results
+	convertedClusterRole := &rbac.ClusterRole{}
+	if err := authorizationapi.Convert_api_ClusterRole_To_rbac_ClusterRole(originClusterRole, convertedClusterRole, nil); err != nil {
+		return nil, err
+	}
+	// do a deep copy here since conversion does not guarantee a new object.
+	equivalentClusterRole := &rbac.ClusterRole{}
+	if err := rbac.DeepCopy_rbac_ClusterRole(convertedClusterRole, equivalentClusterRole, cloner); err != nil {
+		return nil, err
+	}
+
+	// normalize rules before persisting so RBAC's case sensitive authorizer will work
+	normalizePolicyRules(equivalentClusterRole.Rules)
+
+	// there's one wrinkle.  If `openshift.io/reconcile-protect` is to true, then we must set rbac.authorization.kubernetes.io/autoupdate to false to
+	if equivalentClusterRole.Annotations["openshift.io/reconcile-protect"] == "true" {
+		equivalentClusterRole.Annotations["rbac.authorization.kubernetes.io/autoupdate"] = "false"
+		delete(equivalentClusterRole.Annotations, "openshift.io/reconcile-protect")
+	}
+
+	// resource version cannot be set during creation
+	equivalentClusterRole.ResourceVersion = ""
+
+	return equivalentClusterRole, nil
+}
+
+func PrepareForUpdateClusterRole(equivalentClusterRole, rbacClusterRole *rbac.ClusterRole) bool {
```

Comment:
godoc, particularly the mutation bit.  

`(newClusterRole, existingClusterRole *rbac.ClusterRole)`